docs/pages/guides/config-plugins.md

```diff
@@ -0,0 +1,614 @@
+};
+
+// 💡 Usage:
+
+/// Create a config
+const config = {
+  name: 'my app',
+};
+
+/// Use the plugin
+export default withMySDK(config, { apiKey: 'X-XXX-XXX' });
+```
+
+### Importing plugins
+
+You may want to create a plugin in a different file, here's how:
+
+- The root file can be any JS file or a file named `app.plugin.js` in the [root of a Node module](#root-app.plugin.js).
+- The file should export a function that satisfies the [`ConfigPlugin`][configplugin] type.
+- Plugins should be transpiled for Node environments ahead of time!
+  - They should support the versions of Node that [Expo supports](https://docs.expo.io/get-started/installation/#requirements) (LTS).
+  - No `import/export` keywords, use `module.exports` in the shipped plugin file.
+  - Expo only transpiles the user's initial `app.config` file, anything more would require a bundler which would add too many "opinions" for a config file.
+
+Consider the following example that changes the config name:
+
+```
+╭── app.config.js ➡️ Expo Config
+╰── my-plugin.js ➡️ Our custom plugin file
+```
+
+`my-plugin.js`
+
+```js
+module.exports = function withCustomName(config, name) {
+  // Modify the config
+  config.name = 'custom-' + name;
+  // Return the results
+  return config;
+};
+```
+
+`app.config.json`
+
+```json
+{
+  "name": "my-app",
+  "plugins": ["./my-plugin", "app"]
+}
+```
+
+↓ ↓ ↓
+
+**Evaluated config JSON**
+
+```json
+{
+  "name": "custom-app",
+  "plugins": ["./my-plugin", "app"]
+}
+```
+
+### Chaining plugins
+
+Once you add a few plugins, your `app.config.js` code can become difficult to read and manipulate. To combat this, `@expo/config-plugins` provides a `withPlugins` function which can be used to chain plugins together and execute them in order.
+
+```js
+/// Create a config
+const config = {
+  name: 'my app',
+};
+
+// ❌ Hard to read
+withDelta(withFoo(withBar(config, 'input 1'), 'input 2'), 'input 3');
+
+// ✅ Easy to read
+import { withPlugins } from '@expo/config-plugins';
+
+withPlugins(config, [
+  [withBar, 'input 1'],
+  [withFoo, 'input 2'],
+  // When no input is required, you can just pass the method...
+  withDelta,
+]);
+```
+
+To support JSON configs, we also added the `plugins` array which just uses `withPlugins` under the hood.
+Here is the same config as above, but even simpler:
+
+```js
+export default {
+  name: 'my app',
+  plugins: [
+    [withBar, 'input 1'],
+    [withFoo, 'input 2'],
+    [withDelta, 'input 3'],
+  ],
+};
+```
+
+## What are mods
+
+An async function which accepts a config and a data object, then manipulates and returns both as an object.
+
+Modifiers (mods for short) are added to the `mods` object of the Expo config. The `mods` object is different to the rest of the Expo config because it doesn't get serialized after the initial reading, this means you can use it to perform actions _during_ code generation. If possible, you should attempt to use basic plugins instead of mods as they're simpler to work with.
```

Comment:
```suggestion
Mods are added to the `mods` object of the Expo config. The `mods` object is different to the rest of the Expo config because it doesn't get serialized after the initial reading, this means you can use it to perform actions _during_ code generation. If possible, you should attempt to use basic plugins instead of mods as they're simpler to work with.
```